modules/p2/ding_interests/ding_interests.module

```diff
@@ -0,0 +1,652 @@
+      if (empty($field[0]['taxonomy_term'])) {
+        $status_term = taxonomy_term_load($field[0]['tid']);
+      }
+      else {
+        $status_term = $field[0]['taxonomy_term'];
+      }
+      $status = $status_term->tid;
+    }
+    
+    // Show the term if its part of "På baggrund af dine lån foreslår vi
+    // emneord".
+    // And if the term is "Udvalgt", which is a field set on the term page.
+    if ($status == variable_get('ding_interests_selected_tid') && drupal_is_front_page() || !drupal_is_front_page() || isset($variables['content']['show_anyway'])) {
+      $account = $variables['user'];
+      $link = l($variables['term_name'], "taxonomy/term/" . $variables['tid']);
+      if (arg(2) == $variables['tid']) {
+        // No need to link to the term when you're already on the term page:
+        $link = $variables['term_name'];
+      }
+      $variables['content']['ding_interest_teaser_text'] = array(
+        '#type' => 'markup',
+        '#markup' => "<div class='teaser-text'>" .
+        t("Are you interested in more content about") .
+        "</div><div class='term-link'>" .
+        $link .
+        "?</div>",
+        '#weight' => -1,
+      );
+
+      $list = ding_list_get_list(DING_INTERESTS_LIST, $account);
+
+      $value = ding_interest_get_term_cql($term);
+      
+      if (!ding_list_has_ding_list_element($list, 'taxonomy_term', $value)) {
+        $interest_flag = ding_interests_ajax_button($term, $account, DING_INTERESTS_INTERESTING);
+        $no_interest_flag = ding_interests_ajax_button($term, $account, DING_INTERESTS_NOT_INTERESTING);
+        $variables['content']['ding_interest_interest_button'] = array(
+          '#type' => 'markup',
+          '#markup' => $interest_flag,
+          '#weight' => 50,
+        );
+        $variables['content']['ding_interest_no_interest_button'] = array(
+          '#type' => 'markup',
+          '#markup' => $no_interest_flag,
+          '#weight' => 50,
+        );
+      }
+      else {
+        $variables['content']['ding_interest_no_interest_button'] = array(
+          '#type' => 'markup',
+          '#markup' => t('Already added to !list', array('!list' => l(t('interests'), 'list/' . $list->id))),
+          '#weight' => 50,
+        );
+      }
+    }
+  }
+  elseif ($view_mode == 'teaser') {
+    drupal_add_css(drupal_get_path('module', 'ding_interests') . '/ding_interests.css');
+    $variables['classes_array'][] = 'no-content';
+  }
+}
+
+/**
+ * Ajax callback for the add to list button.
+ *
+ * @ajax boolean
+ *   Indicates if this call is running through Ajax.
+ *
+ * @term object taxonomy_term
+ *   The taxonomy term being added.
+ *
+ * @account object user
+ *   The user to act on.
+ *
+ * @return array
+ *   if ajax is used an array of ajax commands is returned,
+ *   else an empty string is returned.
+ */
+function ding_interests_add_to_list_callback($ajax, $term_id, $account, $rating = 5) {
```

Comment:
There does not seem to be any check that `$account` is either the current user or one with sufficient privileges. In practices this means that any user will be able to mark a term as interesting for another user by changing the url.